src/layout/Menu/Menu.minors.tsx

```diff
@@ -26,66 +27,136 @@ export function Section({ children, title = null, ...props }) {
   );
 }
 
-export function Item({
-  action,
-  active,
-  children,
-  href,
-  toggle = false,
-  icon,
-  ...props
-}) {
-  const [open, setOpen] = useState(!toggle);
-
-  const doToggle = () => toggle && setOpen((open) => !open);
+type ItemProps = {
+  action;
+  active;
+  children;
+  href;
+  icon;
+  paddingIncrement;
+  toggle;
+};
 
-  const doAction = (e: MouseEvent) => {
-    action.onClick && action.onClick(e);
-  };
-
-  const simpleKids =
+export function Item({ children, ...props }: ItemProps) {
+  const simple =
     typeof children === 'object' && !children.props
       ? children.flat().filter((child) => typeof child === 'string')
       : children;
 
-  let complexKids =
+  const complex =
     typeof children === 'object' &&
     !children.props &&
     children.flat().filter((child) => typeof child !== 'string');
 
-  if (complexKids.length === 0) complexKids = false;
+  if (complex || props.toggle) {
+    const children = { simple, complex };
+    return <ComplexItem {...props}>{children}</ComplexItem>;
+  }
+
+  if (simple) {
+    const children = simple;
+    return <SimpleItem {...props}>{children}</SimpleItem>;
+  }
+}
+
+const PADDING_INCREMENT = 8;
 
-  const link = simpleKids && href;
-  const $height = open && complexKids.length;
+function SimpleItem({
+  action,
+  active,
+  children,
+  href,
+  icon,
+  paddingIncrement = PADDING_INCREMENT,
+  ...props
+}) {
+  const onClick = (e: MouseEvent) => action.onClick?.(e);
 
   return (
-    <Wrapper active={active} $height={$height}>
+    <Wrapper>
       <ItemStyled
-        onClick={doToggle}
-        as={link ? 'a' : 'button'}
+        active={active}
+        as={href ? 'a' : 'button'}
         href={href}
+        hasAction={Boolean(action)}
+        indentation={paddingIncrement}
         {...props}
       >
         {action && (
-          <Action onClick={doAction}>
+          <Action onClick={onClick}>
             {action.icon}
             <Focus parent={Action} />
           </Action>
         )}
 
         {icon && icon}
-        {simpleKids}
+        <TextContainer>{children}</TextContainer>
         <Focus parent={ItemStyled} />
       </ItemStyled>
+    </Wrapper>
+  );
+}
 
+function ComplexItem({
```

Comment:
`ComplexItem` is getting kind of complex 😆. It **_might_** benefit from the code within the `{toggle &&` block and the `{open &&` block being extrapolated to their own components.